src/software/logger/coloured_cout_sink.cpp

```diff
@@ -4,6 +4,7 @@
 
 ColouredCoutSink::ColouredCoutSink(bool print_detailed) : print_detailed(print_detailed)
 {
+    num_repeats = 0;
 }
```

Comment:
I think in C++, using an initialization list is usually preferred. 
```suggestion
ColouredCoutSink::ColouredCoutSink(bool print_detailed) : print_detailed(print_detailed), num_repeats(0) {}
```